src/celeritas/field/CartMapField.hh

```diff
@@ -0,0 +1,63 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/CartMapField.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Config.hh"
+
+#if CELERITAS_USE_COVFIE
+#    include "detail/CartMapField.covfie.hh"
+#else
+
+#    include "corecel/Assert.hh"
+#    include "corecel/Macros.hh"
+#    include "corecel/Types.hh"
+#    include "corecel/cont/Array.hh"
+
+#    include "CartMapFieldData.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Dummy class for cartesian map magnetic field when no backend is available.
+ */
+class CartMapField
```

Comment:
If we might have more Covfie types in the future, we could define this in a detail header as `template<template<Ownership, MemSpace> class P> class NotImplementedField` and simply alias it here. That would also make this file a bit more readable.